File: Protogen/loxim_protocol/java/src/main/java/pl/edu/mimuw/loxim/protocol/packages_data/BobPackage.java
package pl.edu.mimuw.loxim.protocol.packages_data;

import java.util.EnumSet;
import pl.edu.mimuw.loxim.protogen.lang.java.template.ptools.Package;
import pl.edu.mimuw.loxim.protogen.lang.java.template.exception.ProtocolException;
import pl.edu.mimuw.loxim.protogen.lang.java.template.streams.PackageInputStreamReader;
import pl.edu.mimuw.loxim.protogen.lang.java.template.streams.PackageOutputStreamWriter;


import pl.edu.mimuw.loxim.protocol.enums.*;


public  class BobPackage extends Package {
		public final static long ID=(long)15;
	
    	/**
**/
    	private String value=null;
    	
	public BobPackage()
	{
	}

		public BobPackage(
		    	String a_value    	)
	{
        	value=a_value;
    
	}
		
		@Override
	public long getPackageType() {
		return (long)ID;
	}
		
	@Override
	protected void deserializeW(PackageInputStreamReader reader)
			throws ProtocolException {
		super.deserializeW(reader);
		
				try{
	    			value=reader.readString();
				}catch(Exception e)
		{
			throw new ProtocolException(e);
		}
			}
	
	@Override
	protected void serializeW(PackageOutputStreamWriter writer)
			throws ProtocolException {
		super.serializeW(writer);
				try{
	    			writer.writeString(value);		
				}catch(Exception e)
		{
			throw new ProtocolException(e);
		}	
			}
	
	@Override
	public boolean equals(Object obj) {
		if(!super.equals(obj))
			return false;
		BobPackage p=(BobPackage)obj;
	    if (!(((value==p.getValue())||((value!=null)&&(p.getValue()!=null)&&((value).equals(p.getValue())))))) 
return false;
				return true;
	}
	
//================= GETTERS AND SETTERS ===================

        	public String getValue(){return value;};
		
    	public void setValue(String a_value){ value=a_value;};
		
    
	 
}

File: Protogen/loxim_protocol/cpp/protocol/packages_data/BobPackage.cpp
#include <string.h>
#include <stdlib.h>
#include "BobPackage.h"
#include "ptools/PackageBufferWriter.h"
#include "ptools/PackageBufferReader.h"

using namespace protocol;

BobPackage::BobPackage(\
		CharArray* a_value
		)
{
	value=a_value;
}

BobPackage::BobPackage()
{
	value=NULL;
}

BobPackage::~BobPackage()
{
	if (value) delete value;
}

void BobPackage::serializeW(PackageBufferWriter *writer)
{
	Package::serializeW(writer);
	writer->writeString(value);
}

bool BobPackage::deserializeR(PackageBufferReader *reader)
{
	if (!Package::deserializeR(reader)) return false;
	value=reader->readString();
	return true;
}

bool BobPackage::equals(Package *_package)
{
	if (!_package) return false;
	if (_package->getPackageType()!=this->getPackageType()) return false;
	if (!Package::equals(_package)) return false;
	BobPackage* p=(BobPackage*)_package;
	if (!(CharArray::equal(getValue(),p->getValue()))) return false;
	return true;
}
